## Data Science Case Study on Airbnb dataset
## Task to perform to complete the given case study


* 1. Import pandas, matplotlib, also using %matplotlib magic command write a the parameter which can print the chart.
* 2. Read the CSV file and create pandas dataframe object airbnb
* 3. Review first 5 rows
* 4. What are the room_type available
* 5. Drop the shared room type as its count is very low compare to others
* 6. Take a count of each neighborhood to show how popluar they are w.r.t to airbnb apartment number.
* 7. Number of reviews that is given for ex.How many property has just 0 reviews or 1 reviews etc.
* 8. What is the overall_statisfaction count by each rating.'
* 9. create a sub dataset where overall satisfaction is 0
* 10. Describle this dataset using describle command and write your observation.
* 11. what is the satisfaction ratings count where bedroom is 3
* 12. what is the satisfaction ratings count where bedroom is 2
* 13. what is satisfaction ratings count where price is greater than equal to 150 but less than equal to 180
* 14. Create a copy of main airbnb dataset and impute the value 4 where ever the satisfaction count is 0
* 15. Now take the count of satisfaction ratings again after this imputation and compare with main dataset where still 0 is present
* 16. Find out the count of each accomodation value to understand where is the most availability.
* 17. On which price most of the properties are available.
* 18 What is the proportion of each bedroom values in overall bedroom values
* 19. Drop the values where bedroom are equal to zero 
* 20. Now take the count of bedroom again which is without the zero values.

In [1]:
#task 1 Import pandas, matplotlib
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
#task 2 Read the CSV file and create pandas dataframe object airbnb
airbnb = pd.read_csv('airbnb prices.csv')

In [4]:
#task 3 Review first 5 rows
airbnb.head()

room_id  survey_id   host_id    room_type  country       city  borough  \
0  10176931       1476  49180562  Shared room      NaN  Amsterdam      NaN   
1   8935871       1476  46718394  Shared room      NaN  Amsterdam      NaN   
2  14011697       1476  10346595  Shared room      NaN  Amsterdam      NaN   
3   6137978       1476   8685430  Shared room      NaN  Amsterdam      NaN   
4  18630616       1476  70191803  Shared room      NaN  Amsterdam      NaN   

              neighborhood  reviews  overall_satisfaction  accommodates  \
0  De Pijp / Rivierenbuurt        7                   4.5             2   
1             Centrum West       45                   4.5             4   
2          Watergraafsmeer        1                   0.0             3   
3             Centrum West        7                   5.0             4   
4   De Baarsjes / Oud West        1                   0.0             2   

   bedrooms  bathrooms  price  minstay  \
0       1.0        NaN  156.0      NaN   
1       1.0        NaN  126.0      NaN   
2       1.0        NaN  132.0      NaN   
3       1.0        NaN  121.0      NaN   
4       1.0        NaN   93.0      NaN   

                                             name               last_modified  \
0        Red Light/ Canal view apartment (Shared)  2017-07-23 13:06:27.391699   
1  Sunny and Cozy Living room in quite neighbours  2017-07-23 13:06:23.607187   
2                                       Amsterdam  2017-07-23 13:06:23.603546   
3                    Canal boat RIDE in Amsterdam  2017-07-23 13:06:22.689787   
4    One room for rent in a three room appartment  2017-07-23 13:06:19.681469   

    latitude  longitude                                           location  
0  52.356209   4.887491  0101000020E610000033FAD170CA8C13403BC5AA41982D...  
1  52.378518   4.896120  0101000020E6100000842A357BA095134042791F477330...  
2  52.338811   4.943592  0101000020E6100000A51133FB3CC613403543AA285E2B...  
3  52.376319   4.890028  0101000020E6100000DF180280638F134085EE92382B30...  
4  52.370384   4.852873  0101000020E6100000CD902A8A57691340187B2FBE682F...

In [5]:
airbnb.tail()

room_id  survey_id    host_id     room_type  country       city  \
18718  17789893       1476   47501089  Private room      NaN  Amsterdam   
18719  16877166       1476   67093870  Private room      NaN  Amsterdam   
18720  19859427       1476   29724632  Private room      NaN  Amsterdam   
18721  17132164       1476  115156569  Private room      NaN  Amsterdam   
18722   7605782       1476   39503013  Private room      NaN  Amsterdam   

       borough             neighborhood  reviews  overall_satisfaction  \
18718      NaN          Bijlmer Centrum       10                   5.0   
18719      NaN          Bijlmer Centrum        6                   5.0   
18720      NaN  Geuzenveld / Slotermeer        0                   0.0   
18721      NaN             Centrum West       13                   4.5   
18722      NaN             Centrum West      113                   4.5   

       accommodates  bedrooms  bathrooms  price  minstay  \
18718             3       1.0        NaN   32.0      NaN   
18719             4       1.0        NaN   24.0      NaN   
18720             1       1.0        NaN   38.0      NaN   
18721             2       1.0        NaN   36.0      NaN   
18722             2       1.0        NaN   35.0      NaN   

                                                name  \
18718  1-3 pers. Cozy Rm AFAS Live, ArenA, ZIGGODOME   
18719               Modern Room by Arena, ZIGGO, HmH   
18720                            Private single room   
18721    City Center studio in Touristic Amsterdam 1   
18722               I have a room available for rent   

                    last_modified   latitude  longitude  \
18718  2017-07-22 16:05:14.158963  52.319794   4.955638   
18719  2017-07-22 16:05:14.151986  52.319080   4.954822   
18720  2017-07-22 16:05:14.149610  52.384028   4.838403   
18721  2017-07-22 16:05:14.146183  52.372120   4.890982   
18722  2017-07-22 16:05:12.257054  52.381392   4.899658   

                                                location  
18718  0101000020E6100000684293C492D2134080BA8102EF28...  
18719  0101000020E61000005801BEDBBCD1134062670A9DD728...  
18720  0101000020E61000002079E750865A1340C85F5AD42731...  
18721  0101000020E6100000774CDD955D9013400118CFA0A12F...  
18722  0101000020E6100000CD565EF23F9913405F7AFB73D130...

In [6]:
#Task 4 What are the room_type available
airbnb.room_type.unique()

array(['Shared room', 'Entire home/apt', 'Private room'], dtype=object)

There are 3 types of room available "Shared room", "Entire Home or Apartment" and "priavte room"

In [7]:
#Task 5 Drop the shared room type as its count is very low compare to others
airbnb = airbnb[airbnb.room_type != 'Shared room']

In [8]:
airbnb.loc[airbnb.room_type == 'Shared room']

Empty DataFrame
Columns: [room_id, survey_id, host_id, room_type, country, city, borough, neighborhood, reviews, overall_satisfaction, accommodates, bedrooms, bathrooms, price, minstay, name, last_modified, latitude, longitude, location]
Index: []

In [9]:
# task 6 Take a count of each neighorhood to show how popluar they are w.r.t to airbnb apartment number.
airbnb.groupby('neighborhood').room_id.count()

neighborhood
Bijlmer Centrum                             98
Bijlmer Oost                                97
Bos en Lommer                              985
Buitenveldert / Zuidas                     248
Centrum Oost                              1724
Centrum West                              2217
De Aker / Nieuw Sloten                     114
De Baarsjes / Oud West                    3280
De Pijp / Rivierenbuurt                   2371
Gaasperdam / Driemond                       42
Geuzenveld / Slotermeer                    192
Ijburg / Eiland Zeeburg                    378
Noord Oost                                 221
Noord West                                 241
Noord-West / Noord-Midden                 1416
Oostelijk Havengebied / Indische Buurt     916
Osdorp                                     161
Oud Noord                                  492
Oud Oost                                  1167
Slotervaart                                346
Watergraafsmeer                            515


In [13]:
# task 7. Number of reviews that is given for ex.How many property has just 0 reviews or 1 reviews etc.
zero_review = airbnb['reviews'] == 0
one_review = airbnb['reviews'] == 1

airbnb.reviews.value_counts()

0      2975
1      1500
2      1238
3      1100
4       924
       ... 
310       1
390       1
447       1
215       1
183       1
Name: reviews, Length: 283, dtype: int64

the properties with 0 reviews are 2975 and 1 reviews are 1500 so on

In [21]:
#task 8 What is the overall_statisfaction count by each rating

airbnb.overall_satisfaction.value_counts()

5.0    7697
0.0    5721
4.5    4543
4.0     573
3.5     105
3.0      19
2.5       1
1.0       1
Name: overall_satisfaction, dtype: int64

In [25]:
#task 9 create a sub dataset where overall satisfaction is 0
sub_dataset = airbnb[airbnb.overall_satisfaction == 0]

In [26]:
sub_dataset

room_id  survey_id    host_id     room_type  country       city  \
2      14011697       1476   10346595   Shared room      NaN  Amsterdam   
4      18630616       1476   70191803   Shared room      NaN  Amsterdam   
7      19590049       1476  132687356   Shared room      NaN  Amsterdam   
8       5020280       1476    4059485   Shared room      NaN  Amsterdam   
9      15810783       1476   84978218   Shared room      NaN  Amsterdam   
...         ...        ...        ...           ...      ...        ...   
18708  16411584       1476   83673574  Private room      NaN  Amsterdam   
18709  12886265       1476   50843183  Private room      NaN  Amsterdam   
18714  19203256       1476  132265798  Private room      NaN  Amsterdam   
18715  19734178       1476  139135665  Private room      NaN  Amsterdam   
18720  19859427       1476   29724632  Private room      NaN  Amsterdam   

       borough                            neighborhood  reviews  \
2          NaN                         Watergraafsmeer        1   
4          NaN                  De Baarsjes / Oud West        1   
7          NaN                              Westerpark        2   
8          NaN                                Oud Oost        2   
9          NaN                            Centrum West        0   
...        ...                                     ...      ...   
18708      NaN  Oostelijk Havengebied / Indische Buurt        0   
18709      NaN               Noord-West / Noord-Midden        0   
18714      NaN                         Bijlmer Centrum        1   
18715      NaN                                  Osdorp        0   
18720      NaN                 Geuzenveld / Slotermeer        0   

       overall_satisfaction  accommodates  bedrooms  bathrooms  price  \
2                       0.0             3       1.0        NaN  132.0   
4                       0.0             2       1.0        NaN   93.0   
7                       0.0             2       1.0        NaN  414.0   
8                       0.0             2       1.0        NaN  222.0   
9                       0.0            12       1.0        NaN  301.0   
...                     ...           ...       ...        ...    ...   
18708                   0.0             1       1.0        NaN   37.0   
18709                   0.0             1       1.0        NaN   28.0   
18714                   0.0             4       1.0        NaN   35.0   
18715                   0.0             1       0.0        NaN   30.0   
18720                   0.0             1       1.0        NaN   38.0   

       minstay                                               name  \
2          NaN                                          Amsterdam   
4          NaN       One room for rent in a three room appartment   
7          NaN                                   big boot Adam 04   
8          NaN                  Bright modern appartment in East!   
9          NaN     CANAL BOATTOUR AMSTERDAM covered boat 1,5 hour   
...        ...                                                ...   
18708      NaN      Single Room with double bed in Amsterdam east   
18709      NaN                       Small room - Amsterdam south   
18714      NaN  NEW Stylish room, Ziggodome, AFAS LIVE, ArenA,...   
18715      NaN                       Cozy Apartment in Nieuw-West   
18720      NaN                                Private single room   

                    last_modified   latitude  longitude  \
2      2017-07-23 13:06:23.603546  52.338811   4.943592   
4      2017-07-23 13:06:19.681469  52.370384   4.852873   
7      2017-07-23 13:06:09.984748  52.375205   4.866117   
8      2017-07-23 13:06:07.452609  52.357346   4.912887   
9      2017-07-23 13:06:07.447989  52.386610   4.890128   
...                           ...        ...        ...   
18708  2017-07-22 16:05:14.190492  52.362315   4.944151   
18709  2017-07-22 16:05:14.187572  52.346610   4.846674   
18714  2017-07-22 16:05:14.168799  52.320049   4.955609   
18715 

In [33]:
#task 10 Describle this dataset using describle command and write your observation
airbnb.describe()

room_id  survey_id       host_id  country  borough       reviews  \
count  1.866000e+04    18660.0  1.866000e+04      0.0      0.0  18660.000000   
mean   1.120262e+07     1476.0  3.575860e+07      NaN      NaN     16.729207   
std    6.084245e+06        0.0  3.758082e+07      NaN      NaN     33.506653   
min    2.818000e+03     1476.0  2.234000e+03      NaN      NaN      0.000000   
25%    6.047014e+06     1476.0  7.125411e+06      NaN      NaN      2.000000   
50%    1.228306e+07     1476.0  1.987107e+07      NaN      NaN      6.000000   
75%    1.661017e+07     1476.0  5.201818e+07      NaN      NaN     17.000000   
max    2.000373e+07     1476.0  1.418319e+08      NaN      NaN    532.000000   

       overall_satisfaction  accommodates      bedrooms  bathrooms  \
count          18660.000000  18660.000000  18660.000000        0.0   
mean               3.303778      2.921543      1.431833        NaN   
std                2.213003      1.319141      0.880145        NaN   
min                0.000000      1.000000      0.000000        NaN   
25%                0.000000      2.000000      1.000000        NaN   
50%                4.500000      2.000000      1.000000        NaN   
75%                5.000000      4.000000      2.000000        NaN   
max                5.000000     17.000000     10.000000        NaN   

              price  minstay      latitude     longitude  
count  18660.000000      0.0  18660.000000  18660.000000  
mean     166.811897      NaN     52.365261      4.888606  
std      108.961385      NaN      0.015141      0.034530  
min       12.000000      NaN     52.296200      4.763264  
25%      108.000000      NaN     52.355252      4.864372  
50%      144.000000      NaN     52.364623      4.885995  
75%      192.000000      NaN     52.374789      4.907502  
max     6000.000000      NaN     52.424980      5.027689

my obseravation using the describe method is that the total 18660 people have posted their reviews 
and the average overall satisfacation is 3.3 and also the average price of most of the airbnb's is 166.81

In [59]:
# task 11 what is the satisfaction ratings count where bedroom is 3
airbnb.groupby('bedrooms').overall_satisfaction.count()[3:3]

bedrooms
3.0    1444
Name: overall_satisfaction, dtype: int64

The satisfaction ratings count for bedroom 3 is 1444

In [60]:
#task 12 what is the satisfaction ratings count where bedroom is 2
airbnb.groupby('bedrooms').overall_satisfaction.count()[2:2]

bedrooms
2.0    4456
Name: overall_satisfaction, dtype: int64

In [7]:
#task 13 what is satisfaction ratings count where price is greater than equal to 150 but less than equal to 180
new_df = airbnb[(airbnb['price'] >= 150) & (airbnb['price'] <=180 )] 
count = new_df['overall_satisfaction'].count()

In [12]:
count

3809

In [40]:
#task 14 Create a copy of main airbnb dataset and impute the value 4 where ever the satisfaction is count 0
copy_dataset = airbnb[airbnb.overall_satisfaction == 0]

In [49]:
copy_dataset['overall_satisfaction']= 4

/home/akbar/.local/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [51]:
copy_dataset.overall_satisfaction.count()

5748

In [20]:
#Task 15 Now take the count of satisfaction ratings again after this imputation and compare with main dataset where still 0 is present
print(f"Satisfaction count from the original data: {airbnb.overall_satisfaction.count()}")
print(f"Satisfaction count from the sub dataset {copy_dataset.overall_satisfaction.count()}")

Satisfaction count from the original data: 18723
Satisfaction count from the sub dataset 5748


In [95]:
#task 16 Find out the count of each accomodation value to understand where is the most availability.
accomodation = airbnb.accommodates.value_counts()

In [96]:
accomodation.sort_values()

13       1
17       1
11       2
14       6
9        8
12       9
10      16
16      18
7       52
8      105
1      355
5      470
6      475
3     1583
4     5563
2     9996
Name: accommodates, dtype: int64

the most available accomodation is 2 

In [103]:
#task 17 On which price most of the properties are available.
airbnb.groupby('room_type').price.mean()
#airbnb.head()

room_type
Entire home/apt    181.547136
Private room       106.870451
Name: price, dtype: float64

In [104]:
#task 18 What is the proportion of each bedroom values in overall bedroom values
airbnb.head()

room_id  survey_id   host_id        room_type  country       city  \
33  10201214       1476   8497487  Entire home/apt      NaN  Amsterdam   
34   3119298       1476  15847782  Entire home/apt      NaN  Amsterdam   
35   5372074       1476  14390964  Entire home/apt      NaN  Amsterdam   
36  10511291       1476   5421078  Entire home/apt      NaN  Amsterdam   
37  18047871       1476   9302267  Entire home/apt      NaN  Amsterdam   

    borough             neighborhood  reviews  overall_satisfaction  \
33      NaN  Ijburg / Eiland Zeeburg       19                   5.0   
34      NaN               Westerpark        1                   0.0   
35      NaN             Centrum West       10                   5.0   
36      NaN          Watergraafsmeer        1                   0.0   
37      NaN             Centrum West        0                   0.0   

    accommodates  bedrooms  bathrooms   price  minstay  \
33             8       4.0        NaN   763.0      NaN   
34             8       3.0        NaN   445.0      NaN   
35             4       3.0        NaN   721.0      NaN   
36             8       3.0        NaN  1412.0      NaN   
37             4       4.0        NaN   541.0      NaN   

                                              name  \
33               Beautiful Watervilla in Amsterdam   
34                     Modern 5-8 person apartment   
35                       Prinsengracht Appartement   
36              Whole house 200m2 Frankendael Park   
37  Great 4 bedroom apartment at the flower market   

                 last_modified   latitude  longitude  \
33  2017-07-23 13:02:10.468528  52.348254   5.001477   
34  2017-07-23 12:58:15.945759  52.377581   4.873119   
35  2017-07-23 12:52:40.980471  52.373078   4.884269   
36  2017-07-23 12:30:42.249974  52.350968   4.929821   
37  2017-07-23 12:30:27.637777  52.367890   4.889273   

                                             location  
33  0101000020E6100000E7C8CA2F83011440C0594A96932C...  
34  0101000020E61000009D103AE8127E1340A54BFF925430...  
35  0101000020E61000006C770FD07D891340B56D1805C12F...  
36  0101000020E61000003B8908FF22B81340A5F8F884EC2C...  
37  0101000020E6100000FDDCD0949D8E13404243FF04172F...

In [106]:
#task 19 Drop the values where bedroom are equal to zero
airbnb = airbnb[airbnb.bedrooms != 0]

In [107]:
# task 20 Now take the count of bedroom again which is without the zero values.
airbnb.bedrooms.value_counts()

1.0     11038
2.0      4456
3.0      1444
4.0       473
5.0        62
6.0        19
10.0        5
7.0         4
8.0         3
9.0         2
Name: bedrooms, dtype: int64